sdk/python/kfp/dsl/_pipeline.py

```diff
@@ -19,6 +19,11 @@
 import sys
 
 
+# Contains a stack of handler functions.
+# The last handler in the list is called whenever the @pipeline decorator is used
+_pipeline_decorator_handlers = []
```

Comment:
Maybe we only need one decorator handler? What are the scenarios where multiple decorator handlers are needed?

Reply:
We only call the last one, so we only need one.
But we need to put the old one somewhere.
Having a stack makes it easier as you do not need to remember the old handler - you just call `.pop()`.

If you want, I can replace this handler stack with a single handler.

Reply:
I've switched to single handler.